deps.go

```diff
@@ -0,0 +1,236 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"gopkg.in/errgo.v2/fmt/errors"
+)
+
+func deps(pkgNames []string, withTests bool) (build, test []*Package, err error) {
+	if len(pkgNames) == 0 {
+		// The "no packages means the current package" behaviour
+		// can be done at a higher level.
+		return nil, nil, nil
+	}
+	args := []string{"list", "-deps", "-json"}
+	if withTests {
+		args = append(args, "-test")
+	}
+	args = append(args, pkgNames...)
+	cmd := exec.Command("go", args...)
+	cmd.Stderr = os.Stderr
+	var buf bytes.Buffer
+	cmd.Stdout = &buf
+	if err := cmd.Run(); err != nil {
+		return nil, nil, errors.Wrap(err)
+	}
+	var pkgs []*Package
+	dec := json.NewDecoder(&buf)
+	for {
+		var pkg Package
+		if err := dec.Decode(&pkg); err != nil {
+			if err != io.EOF {
+				return nil, nil, errors.Wrap(err)
+			}
+			break
+		}
+		pkgs = append(pkgs, &pkg)
+	}
+	pkgMap := make(map[string]*Package)
+	for _, p := range pkgs {
+		pkgMap[p.ImportPath] = p
+	}
+	// deps holds a map from package import path to whether that package
+	// is used for testing only (true) or just the build (false)
+	deps := make(map[string]bool)
```

Comment:
perhaps name this `testDepOnly` to be clearer?

Reply:
renamed to testOnly.